### PR TITLE
ES6, Node 4, ESLint 3

### PR DESCRIPTION
**Pull request by kevinrenskers**
Upgrading to ESLint 3 and Node 4. Let's use ES6, no reason not to, and personally I don't want to write code in ES5 anymore :)

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,8 +1,8 @@
 'use strict';
 
-var raml2obj = require('raml2obj');
-var pjson = require('./package.json');
-var Q = require('q');
+const raml2obj = require('raml2obj');
+const pjson = require('./package.json');
+const Q = require('q');
```
Comment (fesor) [inline index.js]: Can't we just use native promises?

Comment (kevinrenskers) [inline index.js]: Good point, added another commit.

File: index.js [hunk 2]
```diff
@@ -102,25 +101,25 @@ function getDefaultConfig(mainTemplate, templatesPath) {
       html = html.replace(/&quot;/g, '"');
 
       // Return the promise with the html
-      return Q.fcall(() => html);
+      return new Promise((resolve) => {
```
Comment (fesor) [inline index.js]:
```
return new Promise((resolve) => resolve(html));
```

??

**Comment by kevinrenskers** [comment]
Maybe we should wait with merging this until after 2.5.0 is released. After all requiring Node 4 is a pretty big change. Then this can be part of 3.0.0.

**Pull request merged**